### PR TITLE
Removed pre-converter fallback in URLRouter.

### PR DESCRIPTION
**Pull request by carltongibson**
Hey @matthiask 👋 — wondering if I can ask you to comment on this? (Been a while I know... 🙂) 

I need to look at _exactly_ what it's doing, but I'm inclined to just inline `route_pattern_match()` now. 🤔
What are your thoughts. 

Thanks! 🎁

### CODE REVIEW

File: channels/routing.py [hunk 1]
```diff
@@ -74,26 +74,13 @@ def route_pattern_match(route, path):
...
-        match = route.pattern.match(path)
-        if match:
-            path, args, kwargs = match
-            kwargs.update(route.default_args)
-            return path, args, kwargs
-        return match
-
-    # Django<2.0. No converters... :-(
-    match = route.regex.search(path)
+    assert hasattr(route, "pattern")
```
Comment (carltongibson) [inline channels/routing.py:77]: This can go. Just left it as a sanity check for the moment.

**Review by matthiask — COMMENTED** [review]
Indeed, it has been a while!

I think the change looks good. I checked the test suite and if the inner router's `kwargs` hold the correct values it should be fine (see https://github.com/django/channels/blob/9c1ac42f2a5b4dbfcdade02bd52445bab2851cb9/tests/test_routing.py#L148-L156 and https://github.com/django/channels/blob/9c1ac42f2a5b4dbfcdade02bd52445bab2851cb9/tests/test_routing.py#L195-L198)

Tests are passing after all and I have more confidence in the tests than in my judgement (even in tests I may have added myself)

The `route_pattern_match` docstring should be changed because no backporting is necessary anymore. Which makes me wonder if `route_pattern_match` could be removed altogether now. I'm 99% sure it has to stay because https://github.com/django/django/blob/00370342ca3a478660372975b3309ffc4d535be1/django/urls/conf.py#L80 sets `is_endpoint` to `True` (since the `URLRouter` is callable) but this is wrong because we want to continue resolving at this point in time.

**Comment by carltongibson** [comment]
Thanks @matthiask, that's really helpful. Let me have a read and a ponder. 😉

File: channels/routing.py [hunk 2]
```diff
@@ -135,9 +108,10 @@ async def __call__(self, scope, receive, send):
         # Run through the routes we have until one matches
         for route in self.routes:
             try:
-                match = route_pattern_match(route, path)
+                match = route.pattern.match(path)
                 if match:
                     new_path, args, kwargs = match
+                    kwargs.update(route.default_args)
```
Comment (carltongibson) [inline channels/routing.py:114]: What's this for? 🤔

**Comment by carltongibson** [comment]
Hey @matthiask — Can I get you to look again? 

I inlined the function. Tests pass... 😬

I wasn't sure what the `kwargs.update()` line was for. (See inline comment) I can investigate (by commenting that out to start with 😜)

File: channels/routing.py (hunk 2 above)
Comment (carltongibson) [reply channels/routing.py:115]: Ok, so the example is this: 

```
            path("defaultkwargs/", defaultkwarg_app, kwargs={"default": 42}),
```

So we're adding the `kwargs={"default": 42}` — added a comment in https://github.com/django/channels/pull/1894/commits/0276b612c695a040f8ee4ffddbf0240e0ff9421c

Comment (matthiask) [reply channels/routing.py:115]: Yes, exactly. The consumer should receive the merged arguments from all patterns *and* from the data passed to `path` (or `re_path`)

**Review by matthiask — APPROVED** [review]
Looks good to me. I had to check if it's possible to pass positional arguments to `path()` (for the view/consumer) but that's not possible.

This is a really nice simplification of the code!

**Comment by carltongibson** [comment]
Thanks @matthiask 🙏

**Pull request merged**